drivers/serial/uart_pl011.c

```diff
@@ -432,6 +480,15 @@ static int pl011_init(const struct device *dev)
 		pl011_disable(dev);
 		pl011_disable_fifo(dev);
 
+		/*
+		 * On Ambiq Apollo4 SoCs clock must be enabled and configured via
+		 * the CR register.
+		 */
+#ifdef CONFIG_UART_PL011_AMBIQ_CLK_CR_REG
+		pl011_enable_clk(dev);
+		pl011_clk_set(dev, config->sys_clk_freq);
```

Comment:
error handling missing here

Reply:
Return value of the clk_set function is handled now (the function has a different name now after implementing vendor quirks): https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/drivers/serial/uart_pl011_ambiq.h#L51